x-pack/plugins/lens/public/datasources/text_based/types.ts

```diff
@@ -21,8 +21,9 @@ export interface TextBasedField {
 }
 
 export interface TextBasedLayer {
-  index: string;
-  query: AggregateQuery | undefined;
+  index?: string;
```

Comment:
when table is provided index and query are optional, which introduces some additional checks in the code for this.